### DISCUSSION TITLE
Resource not accessible by integration

### DISCUSSION THREAD

**Discussion by dpo** [community/community#34415] [Actions] [answered]
Hi. I'm writing an action in which I want to update a gist. The relevant step is as follows:
```yaml
      - name: Update Gist
        run: |
          curl \
            -X PATCH \
            -H "Accept: application/vnd.github+json" \
            -H "Authorization: Bearer ${{ github.token }}" \
            https://api.github.com/gists/${{ secrets.GIST_ID }} \
            -d '"description": "My description", "files": ${{ steps.gistjson.outputs.result }}'
```
The step produces the message
```
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/gists#update-a-gist"
}
```
I generated a token with the gist scope. Do I need anything else? The [documentation](https://docs.github.com/en/rest/gists/gists#authentication) says
> To read or write gists on a user's behalf, you need the gist OAuth scope and a token.

but I'm not really clear on what "the gist OAuth scope **and** a token" really means. I understood a token with the gist scope, but I may be wrong.

Thanks in advance!

**Comment by airtower-luna** [comment, marked as answer]
The step uses the automatically generated `github.token` for the workflow run, not any token you have generated:

```
            -H "Authorization: Bearer ${{ github.token }}" \
```

That token is limited to the current repository, so it can't update Gists. Generating a PAT with **gist** scope should work. You'll need to store the PAT as a secret, and then use that secret instead of `github.token`.

**Reply by dpo** [reply]
Of course, thank you!
